src/error.rs

```diff
@@ -65,6 +65,19 @@ impl Error {
     /// custom errors.
     pub const CUSTOM_START: u32 = (1 << 31) + (1 << 30);
 
+    #[cold]
+    pub(super) fn new_os(code: u32) -> Self {
```

Comment:
In the future we might consider adding a public `new_custom` constructor and then deprecating the `From<NonZeroU32>` that does the non-range-checked conversion.